lightning-invoice/src/utils.rs

```diff
@@ -218,6 +189,95 @@ where
 	}
 }
 
+/// Filters channels for an invoice, and returns the corresponding route hints to include
+/// in the invoice.
+///
+/// The filtering ensures that the `RouteHints` returned only include the highest inbound capacity
+/// channel per counterparty node. If any channel with a higher inbound capacity than the given
+/// `min_inbound_capacity_msat` exists, channels to other counterparty nodes with a lower inbound
+/// capacity than `min_inbound_capacity_msat` will be filtered out, even if they are the highest
+/// inbound capacity channel for that specific counterparty node.
+/// If a mix between mix public and private the channels exists, the function returns no RouteHints,
+/// and the sender will need to look at the public channels to find a path instead.
+///
+/// Input:
+/// `channels`: The channels to filter.
+/// `min_inbound_capacity_msat`: Defines the lowest inbound capacity channels must have to not
+/// be filtered out, if any other channel above that amount exists.
+///
+/// Result:
+/// `Vec<RouteHint>`: The filtered `RouteHints`. If empty, the channels is a mix between public and
+/// private.
+fn filter_channels(channels: Vec<ChannelDetails>, min_inbound_capacity_msat: Option<u64>) -> Vec<RouteHint>{
+	let mut filtered_channels: HashMap<PublicKey, (u64, RouteHint)> = HashMap::new();
+	let min_inbound_capacity = min_inbound_capacity_msat.unwrap_or(0);
+
+	let mut is_public_values: HashSet<bool> = HashSet::new();
```

Comment:
There's never any reason to have a `HashSet<bool>`, the table is on the heap and the pointers and seeds are substantially larger than just storing two bools on the stack here.

Reply:
Ok thanks, I'll replace it with bools instead. It was an attempt to remove some lines of code, but given your feedback I agree that thats just silly :)